pokemon-league-participation-manager/src/test/java/cz/muni/fi/pa165/pokemon/league/participation/manager/dao/PokemonDAOTest.java

```diff
@@ -0,0 +1,168 @@
+package cz.muni.fi.pa165.pokemon.league.participation.manager.dao;
+
+import cz.muni.fi.pa165.pokemon.league.participation.manager.common.PersistenceApplicationContext;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.entities.Pokemon;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.entities.PokemonSpecies;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.entities.Trainer;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.enums.PokemonType;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.util.List;
+import javax.inject.Inject;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceException;
+import javax.transaction.Transactional;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import org.junit.Assert;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author Jiří Medveď 38451
+ */
+@ContextConfiguration(classes = PersistenceApplicationContext.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@Transactional
+public class PokemonDAOTest extends AbstractTestNGSpringContextTests {
+
+    private static EntityManagerFactory emf;
+
+    @Inject
+    private PokemonDAO pokemonDao;
+
+    @Inject
+    private TrainerDAO trainerDao;
+
+    @Inject
+    private PokemonSpeciesDAO pokemonSpeciesDao;
+
+    private Trainer trainer;
+    private PokemonSpecies species;
+    private Pokemon pokemon1;
+    private Pokemon pokemon2;

```

Comment:
Here you can mock everything and should be mocked it will make setUp() method much sorter and easier.